### ISSUE TITLE
Cancelling google sign in on ios gives wrong error

### ISSUE THREAD

**Issue by vonovak** [googlesamples/google-services#426]
### Describe your environment

  * Device: iphone x simulator
  * OS version: ios ~~13~~ sorry, it was 12.4
  * SDK version: google sign in sdk 5.0.0 and 5.0.1
  * [Android] Google Play Services version: not relevant
  
### Describe the problem:

Canceling the sign in process leads to `kGIDSignInErrorCodeUnknown` instead of `kGIDSignInErrorCodeCanceled`. Please NOTE that I am able to sign in successfully if I don't cancel the flow.

#### Steps to reproduce:

  1. integrate sign in
  2. tap login in
  3. tap cancel in the login modal (see 1st screenshot)
  4. the delegate is called with wrong error (see 2nd screenshot)

![Screenshot 2019-10-02 at 11 25 26](https://user-images.githubusercontent.com/1566403/66034113-e1110980-e508-11e9-9f01-35a495fffa39.png)

  
![Screenshot 2019-10-02 at 11 39 28](https://user-images.githubusercontent.com/1566403/66034319-5381e980-e509-11e9-8195-1b0c0974ca55.png)

#### Observed Results:

  * the delegate is called with `kGIDSignInErrorCodeUnknown`
  
#### Expected Results:

  * the delegate is called with `kGIDSignInErrorCodeCanceled` as covered in [docs](https://developers.google.com/identity/sign-in/ios/reference/Enums/GIDSignInErrorCode#kgidsigninerrorcodecanceled)

#### Relevant Code:

  ```
  // just use the sample app
  ```

**Comment by ayan2206** [comment]
same problem here

**Comment by capJavert** [comment]
I have the same problem.

**Comment by vonovak** [comment]
@samtstern you seem to be active around this repo, can you comment on this issue? Thanks

**Comment by casevictor** [comment]
hey 👋 got the same issue even with the new sdk `5.0.1`

**Comment by samtstern** [comment]
Hi all!  Sorry I have not been monitoring this repo closely but I will file a bug with the Google Sign In SDK folks ASAP and link them to this issue.

**Comment by samtstern** [comment]
For future reference, internal issue number 143211691

**Comment by gabriellet** [comment]
Thanks for filing this bug, my team has also encountered this issue. Is this a publicly available issue tracker? If so, can we get a link to this to track bug status/resolution?

**Comment by samtstern** [comment]
@gabriellet no it's not a publicly available issue tracker, I just provided the number in case you're discussing it with any other Googlers you can reference that and they can look it up.  No updates at the moment.

**Comment by samtstern** [comment]
The team has confirmed that this is a regression beginning in `5.0.0` and will do a patch release soon to address it.  Thanks for your patience everyone!

**Comment by casevictor** [comment]
Thanks for following up 👏

**Comment by dmitriy-ios-dev** [comment]
hey 👋 got the same issue even using the GoogleSignIn 5.0.1
and error code -15

```
Error Domain=org.openid.appauth.general Code=-15
"Issued at time is more than 600 seconds before or after the current time"
UserInfo={NSLocalizedDescription=Issued at time is more than 600 seconds before or after the current time}
```

**Comment by samtstern** [comment]
This has been fixed and released in version `5.0.2`

**Comment by dmitriy-ios-dev** [comment]
> This has been fixed and released in version `5.0.2`

This problem don't resolve in 5.0.2
`Domain=org.openid.appauth.general Code=-15 "Issued at time is more than 600 seconds before or after the current time"`
